src/app/core/services/repo-creator.service.ts

```diff
@@ -29,4 +61,25 @@ export class RepoCreatorService {
       })
     );
   }
+
+ /**
+  * If a Session is unavailable (because the repository is missing) attempt to create IF it is
+  * the BugReporting Phase
+  * @param currentPhase the current phase of the session.
+  * @param phaseRepo the name of the specified repository.
+  * @return - Dummy Observable to give the API sometime to propagate the creation of the new repository since
+  *           the API Call used here does not return any response.
+  */
+ private attemptSessionAvailabilityFix(currentPhase: Phase, phaseRepo: string): Observable<any> {
```

Comment:
```suggestion
 private triggerRepoCreation(currentPhase: Phase, phaseRepo: string): Observable<any> {
```

Similarly, we could rename this to make the meaning more obvious.


We could actually remove this method entirely.
We can setup a separate method that serves to assert the `currentPhase` and user role (i.e. perform the error handling in this method). This method can be called from the custom operator.

`this.githubService.createRepository(phaseRepo);` could then be invoked directly from the custom operator after all the erroneous cases are handled. 

What do you think @seanlowjk ?

Reply:
I think it would definitely be better to split these steps! Sounds good to me!

Reply:
So from my understanding, would it look something like 

```typescript
cacheSessionFixPermission(),
this.repoCreatorService.verifyPhaseDetails(currentPhase, phaseRepo),
this.repoCreatorService.attemptRepoCreation(phaseRepo),  
this.repoCreatorService.verifyRepoCreation(phaseOwner, phaseRepo)
```

Where we verify the `currentPhase`, `currentUser.role`  and then, attempt to create the repo based on the logic in the original `attemptRepoCreation` method? ><

Reply:
I actually meant that the error handling could be done within `attemptRepoCreation` , instead of creating a new operator for the error handling. 

In my view, creating a new operator breaks the work too finely.

Reply:
Ahhh I see... 

I had the understanding thay it would be cleaner to split the error handling into another step, so we can just focus on one operator doing the error checking and the other creating the repo

Reply:
Ok, sure, we can go with your current approach.

Reply:
:+1:

Reply:
I have edited the details of this PR above along with the code changes!